### PR TITLE
Fix issue producing interpolated strings from char literals

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/68425

### CODE REVIEW

File: src/Features/CSharp/Portable/ConvertToInterpolatedString/CSharpConvertConcatenationToInterpolatedStringRefactoringProvider.cs [hunk 1]
```diff
@@ -21,7 +21,13 @@ protected override bool SupportsInterpolatedStringHandler(Compilation compilatio
         => compilation.GetTypeByMetadataName("System.Runtime.CompilerServices.DefaultInterpolatedStringHandler") != null;
 
     protected override string GetTextWithoutQuotes(string text, bool isVerbatim, bool isCharacterLiteral)
-        => isVerbatim
-            ? text.Substring("@'".Length, text.Length - "@''".Length)
-            : text.Substring("'".Length, text.Length - "''".Length);
+    {
+        var contents = isVerbatim
+            ? text["@'".Length..^1]
```
Comment (ToddGrun) [inline src/Features/CSharp/Portable/ConvertToInterpolatedString/CSharpConvertConcatenationToInterpolatedStringRefactoringProvider.cs:26]: >? text["@'".Length..^1] [](http://example.com/codeflow?start=12&length=23)

These didn't include the end single quote previously, but now do?

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/ConvertToInterpolatedString/CSharpConvertConcatenationToInterpolatedStringRefactoringProvider.cs:26]: no.  this is the same as before, just with clearer syntax.  it's saying "take off two characters from the front, and one from the back"

Comment (ToddGrun) [reply src/Features/CSharp/Portable/ConvertToInterpolatedString/CSharpConvertConcatenationToInterpolatedStringRefactoringProvider.cs:26]: Oops, my brain read that wrong.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**